lib/segment/src/vector_storage/async_raw_scorer.rs

```diff
@@ -169,16 +156,20 @@ where
         // Instead of silently falling back to the sync implementation, we prefer to panic
         // and notify the user that they better use the default IO implementation.
 
-        pq.into_sorted_vec()
+        Ok(pq.into_sorted_vec())
```

Comment:
should we check flag one more time before returning? If operation was cancelled we might have partial result. Since we make it explicit, it might be worth maintaining the declared guarantees